.github/workflows/test.yml

```
@@ -21,26 +21,30 @@ jobs:
           bundler-cache: true
       - name: Run RuboCop
         run: bundle exec rake rubocop
+  setup_matrix:
+    name: 'Determine Ruby versions'
+    runs-on: ubuntu-latest
+    outputs:
+      ruby: ${{ steps.ruby.outputs.versions }}
+    steps:
+      - id: ruby
+        uses: voxpupuli/ruby-version@v0
   test:
```

I'd visually separate the groups
```suggestion

  test:
```